src/spikeinterface/qualitymetrics/quality_metric_calculator.py

```diff
@@ -86,7 +86,12 @@ def _select_extension_data(self, unit_ids):
         new_data = dict(metrics=new_metrics)
         return new_data
 
-    def _run(self, verbose=False, **job_kwargs):
+    def _merge_extension_data(self, units_to_merge, new_unit_ids, new_sorting_analyzer):
+        metrics = self._compute_metrics(new_sorting_analyzer)
+        new_data = dict(metrics=metrics)
+        return new_data
```

Comment:
So now this recomputes everything regardless of the `units_to_merge`?

Reply:
Now this has been improved, and metrics are recomputed only for merged indices. Other values are taken from already computed metrics

Reply:
Parfait!